### PR TITLE
MGMT-16831: Added ResourceSync Owner on Fleets

### PR DESCRIPTION
**Pull request by sagidayan**
- Adds owner to fleet if managed by resourcesync
 - block updating fleets via API if owned by a resource
 - when resourcesync is deleted, remove all ownerRefs from fleets
 - when a fleet definition no longer exists in repo, delete the fleet
 - refactor the status of resourcesync making it cleaner.

### CODE REVIEW

File: api/v1alpha1/conditions.go
Comment (avishayt) [inline api/v1alpha1/conditions.go:113]: I understand your motivation here, but technically, if a Condition isn't set, its value should be assumed to be "Unknown."

File: internal/store/fleet.go
Comment (avishayt) [inline internal/store/fleet.go:31]: Can we merge these?

File: internal/store/fleet.go
Comment (avishayt) [inline internal/store/fleet.go:126]: This is just a special case of List().  See my changes here in internal/store/common.go:

https://github.com/flightctl/flightctl/pull/90/files#diff-1c6e9eca029b27fdaae737726193fe389fab0692a101f2e467fc6a4fc59871a3

File: internal/store/resourcesync.go
Comment (avishayt) [inline internal/store/resourcesync.go:104]: Can we simplify this by removing the owner for all fleets in the org where the owner is "ResourceSync/?"

File: internal/store/resourcesync.go
Comment (avishayt) [inline internal/store/resourcesync.go:122]: Use util.SetResourceOwner()

File: internal/store/resourcesync.go
Comment (avishayt) [inline internal/store/resourcesync.go:218]: Use util.SetResourceOwner()

File: internal/tasks/resourcesync.go
Comment (avishayt) [inline internal/tasks/resourcesync.go:85]: Use util.SetResourceOwner()

File: internal/store/fleet.go
Comment (sagidayan) [reply internal/store/fleet.go:31]: Yes, we can :)

File: internal/store/fleet.go
Comment (sagidayan) [reply internal/store/fleet.go:126]: Ill wait for your PR to be merged

File: internal/tasks/resourcesync.go
Comment (avishayt) [inline internal/tasks/resourcesync.go:111]: Do we want to delete the fleet or just remove the owner?  Maybe only delete the fleet if it has no matching devices?

Comment (sagidayan) [reply internal/tasks/resourcesync.go:111]: as we discussed. if the resourcesync is deleted - we remove the owner from all owned resources (fleets). If a definition was removed, we expect the fleet to be removed as well.

But we can revisit this

File: internal/store/resourcesync.go
Comment (sagidayan) [reply internal/store/resourcesync.go:104]: We can, The only thing is that this will bypass the fleet store. Bypassing callbacks/validations  (whatever) when we add them, etc.

File: api/v1alpha1/conditions.go
Comment (sagidayan) [reply api/v1alpha1/conditions.go:113]: Fixed it in the task. Keeping these functions untouched.

File: internal/service/fleet.go
Comment (hexfusion) [inline internal/service/fleet.go:131]: Seems this might be good to log?

File: internal/store/fleet.go
Comment (hexfusion) [inline internal/store/fleet.go:30]: I am surprised that we need to pass gorm into the store. The reason for the store is to abstract away the underlying ORM.

File: internal/service/fleet.go
Comment (hexfusion) [inline internal/service/fleet.go:147]: add log?

File: internal/store/fleet.go
Comment (hexfusion) [reply internal/store/fleet.go:30]: Could we wrap this in a Type maybe something like.

type Transaction(func(tx *gorm.DB))

then interface can be

```
UnsetOwnerForMatchingFleets(ctx context.Context, Transaction, orgId uuid.UUID, owner string) error
```

**Review by hexfusion — APPROVED** [review]
lgtm defer to @avishayt

**Pull request merged**
